gradle-failure-reports/src/main/java/com/palantir/gradle/failurereports/FailureReportFlowAction.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.failurereports;
+
+import java.io.File;
+import org.gradle.api.flow.BuildWorkResult;
+import org.gradle.api.flow.FlowAction;
+import org.gradle.api.flow.FlowParameters;
+import org.gradle.api.provider.Property;
+import org.gradle.api.services.ServiceReference;
+import org.gradle.api.tasks.Input;
+
+public final class FailureReportFlowAction implements FlowAction<FailureReportFlowAction.Parameters> {
+
+    interface Parameters extends FlowParameters {
+        @Input
+        Property<BuildWorkResult> getBuildResult();
+
+        @Input
+        Property<File> getOutputFile();
+
+        @ServiceReference
+        Property<CompileFailuresService> getCompileFailuresService();
```

Comment:
I was wondering how you would be able to access `Task`s using this `FlowAction` interface as Gradle have started banning access to `Project` level mutable collections in new apis... I guess the build service packages that up and you can just use that? This seems like it would be useful for other places we've had this kinda problem.

Reply:
Actually, even with the previous code we never needed access to this from the finalizer task, we just added the stuff we needed as input properties...